scripts/data/lib/parsers/Extractor.ts

```diff
@@ -0,0 +1,25 @@
+import { z, ZodTypeAny } from "zod";
+
+type Patterns<T> = { [key: string]: (doc: T) => any };
+type Values<T, P extends Patterns<T>> = {
+    [key in keyof P]: string extends ReturnType<P[key]> ? string : ReturnType<P[key]>;
+};
+
+export abstract class Extractor<Z extends ZodTypeAny> {
```

Comment:
Classes are so "cringe" 😬

Reply:
haha. prefer me to convert to functions?
What about the model classes? Same cringe feeling?

Reply:
Most likely.. but meh, its not a blocker. Was just a comment :P